packages/flutter/lib/src/painting/alignment.dart

```diff
@@ -0,0 +1,611 @@
+  static const Alignment centerLeft = const Alignment(-1.0, 0.0);
+
+  /// The center point, both horizontally and vertically.
+  static const Alignment center = const Alignment(0.0, 0.0);
+
+  /// The center point along the right edge.
+  static const Alignment centerRight = const Alignment(1.0, 0.0);
+
+  /// The bottom left corner.
+  static const Alignment bottomLeft = const Alignment(-1.0, 1.0);
+
+  /// The center point along the bottom edge.
+  static const Alignment bottomCenter = const Alignment(0.0, 1.0);
+
+  /// The bottom right corner.
+  static const Alignment bottomRight = const Alignment(1.0, 1.0);
+
+  @override
+  AlignmentGeometry add(AlignmentGeometry other) {
+    if (other is Alignment)
+      return this + other;
+    return super.add(other);
+  }
+
+  /// Returns the difference between two [Alignment]s.
+  Alignment operator -(Alignment other) {
+    return new Alignment(x - other.x, y - other.y);
+  }
+
+  /// Returns the sum of two [Alignment]s.
+  Alignment operator +(Alignment other) {
+    return new Alignment(x + other.x, y + other.y);
+  }
+
+  /// Returns the negation of the given [Alignment].
+  @override
+  Alignment operator -() {
+    return new Alignment(-x, -y);
+  }
+
+  /// Scales the [Alignment] in each dimension by the given factor.
+  @override
+  Alignment operator *(double other) {
+    return new Alignment(x * other, y * other);
+  }
+
+  /// Divides the [Alignment] in each dimension by the given factor.
+  @override
+  Alignment operator /(double other) {
+    return new Alignment(x / other, y / other);
+  }
+
+  /// Integer divides the [Alignment] in each dimension by the given factor.
+  @override
+  Alignment operator ~/(double other) {
+    return new Alignment((x ~/ other).toDouble(), (y ~/ other).toDouble());
+  }
+
+  /// Computes the remainder in each dimension by the given factor.
+  @override
+  Alignment operator %(double other) {
+    return new Alignment(x % other, y % other);
+  }
+
+  /// Returns the offset that is this fraction in the direction of the given offset.
+  Offset alongOffset(Offset other) {
+    final double centerX = other.dx / 2.0;
+    final double centerY = other.dy / 2.0;
+    return new Offset(centerX + x * centerX, centerY + y * centerY);
+  }
+
+  /// Returns the offset that is this fraction within the given size.
+  Offset alongSize(Size other) {
+    final double centerX = other.width / 2.0;
+    final double centerY = other.height / 2.0;
+    return new Offset(centerX + x * centerX, centerY + y * centerY);
+  }
+
+  /// Returns the point that is this fraction within the given rect.
+  Offset withinRect(Rect rect) {
+    final double halfWidth = rect.width / 2.0;
+    final double halfHeight = rect.height / 2.0;
+    return new Offset(
+      rect.left + halfWidth + x * halfWidth,
+      rect.top + halfHeight + y * halfHeight,
+    );
+  }
+
+  /// Returns a rect of the given size, centered at this fraction of the given rect.
```

Comment:
I think "centered" is the wrong word here. If it was centered, then a 100x100 square in the top right would be from -50,-50 to 50,50, not from 0,0 to 100,100.

Reply:
Done.